app/main.py

```diff
@@ -1,6 +1,21 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    storage = []
+
+    def inner(*args: Any) -> Any:
+        for item in storage:
+            if item["arguments"] == args:
+                print("Getting from cache")
+                return item["result"]
```

Comment:
Why don't you just change `storage` to a dictionary and store `arguments: result there`?